credentials/alts/alts_test.go

```diff
@@ -287,6 +296,50 @@ func (s) TestCheckRPCVersions(t *testing.T) {
 	}
 }
 
+func TestFullHandshake(t *testing.T) {
+	// If GOMAXPROCS is set to less than 2, do not run this test. This test
+	// requires at least 2 goroutines to succeed (one goroutine where a
+	// server listens, another goroutine where a client runs).
+	if runtime.GOMAXPROCS(0) < 2 {
+		return
+	}
+
+	// Start the fake handshaker service and the server.
+	var wait sync.WaitGroup
+	defer wait.Wait()
+	stopHandshaker, handshakerAddress := startFakeHandshakerService(t, &wait)
+	defer stopHandshaker()
+	stopServer, serverAddress := startServer(t, handshakerAddress, &wait)
+	defer stopServer()
+
+	// Connect to the server, authenticating with ALTS and the fake handshaker service.
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+	vmOnGCP = true
+	creds := NewClientCreds(&ClientOptions{HandshakerServiceAddress: handshakerAddress})
+	dialOptions := []grpc.DialOption{
+		grpc.WithTransportCredentials(creds),
+		grpc.WithBlock(),
+		grpc.WithReturnConnectionError(),
```

Comment:
Are these two dial options necessary? We recommend our users to not use these dial options. So, we shouldn't be using them in our tests unless there is a strong reason for the same.

And if we don't have a good reason to use these dial options, and we can get rid of them, then we can use `grpc.Dial` instead of `grpc.DialContext`.

Reply:
SG, removed and switched to `grpc.Dial`.